
syntax = "proto3";

import "google/protobuf/empty.proto";

package sse.spirt;

service Spirt {

// Setup
rpc setup (SetupMessage) returns (google.protobuf.Empty) {}

// Search
rpc search (SearchRequestMessage) returns (stream SearchReply) {}
rpc Rsearch (stream SearchRequestMessage) returns (stream SearchReply) {}

// Update
rpc update (UpdateRequestMessage) returns (google.protobuf.Empty) {}
rpc bulk_update (stream UpdateRequestMessage) returns (google.protobuf.Empty) {}

}

message SetupMessage
{
    uint64 setup_size = 1;
    // bytes public_key = 2;
}

message SearchRequestMessage
{
    bytes search_token = 1;
    bytes derivation_key = 2;
    fixed32 add_count = 3;
    bytes key = 4;
}

message SearchReply
{
    bytes result = 1;
}

message UpdateRequestMessage
{
    bytes update_token = 1;
    //uint64 index = 2;
    bytes index = 2;
}
